### PR TITLE
Allow passing a function for error message

### PR DESCRIPTION
**Pull request by apalm**
This is basically a straight port of https://github.com/25th-floor/revalidation/pull/6.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -24,16 +24,19 @@ const transform = (successFn: Function, failFn: Function, input: Array<any>): an
...
  * @param {Object} inputs the input object - in case the predicate function needs access to dependent values
  * @returns {Boolean}
  */
 const runPredicate = ([predicate, errorMsg]:[Function, string],
   value:any,
-  inputs:Object) => predicate(value, inputs)
-  ? true
-  : errorMsg
+  inputs:Object) => {
+  if (predicate(value, inputs)) {
```
Comment (apalm) [inline src/index.js]: I changed it to an if statement because of ESLint's `no-nested-ternary` rule.

Comment (busypeoples) [inline src/index.js]: Maybe we should disable `no-nested-ternary` rule? 
I think we can write this in a readable manner anyway.
`
() => predicate(value, inputs)
  ? true
  : typeof errorMsg === 'function' 
  ? errorMsg(value) 
  : errorMsg
`
Any thoughts on this?

Comment (apalm) [inline src/index.js]: @busypeoples Sure? That's how I wrote it originally.

Did you mean disabling the rule in the config or just here with a `eslint-disable-line`?

Comment (busypeoples) [inline src/index.js]: Disabling it here for a start, should be enough for now. Great work and input by the way!

**Comment by apalm** [comment]
@busypeoples Updated. Let me know if you want me to squash the commits or do anything else.

**Comment by busypeoples** [comment]
@apalm Thanks! Sure, squash the commits. I will merge it then.

**Comment by busypeoples** [comment]
Resolves #2

**Pull request merged**